### PR TITLE
simplified reconcilerbase constructor

### PR DESCRIPTION
**Pull request by raffaelespazzoli**
added utility functions for direct client
simplified reconcilerbase constructor

Signed-off-by: raffaelespazzoli <raffaele.spazzoli@gmail.com>

### CODE REVIEW

File: pkg/util/reconciler.go [hunk 1]
```diff
@@ -59,12 +65,19 @@ type ReconcilerBase struct {
 // 		ReconcilerBase: util.NewReconcilerBase(mgr.GetClient(), mgr.GetScheme(),mgr.GetConfig()),
 // 	}
 // }
-func NewReconcilerBase(client client.Client, scheme *runtime.Scheme, restConfig *rest.Config, recorder record.EventRecorder) ReconcilerBase {
+// func NewReconcilerBase(client client.Client, scheme *runtime.Scheme, restConfig *rest.Config, recorder record.EventRecorder) ReconcilerBase {
```
Comment (sabre1041) [inline pkg/util/reconciler.go:68]: Can we clean up these comments?

File: pkg/util/reconciler.go [hunk 2]
```diff
@@ -46,10 +51,11 @@ import (
 type ReconcilerBase struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	client     client.Client
-	scheme     *runtime.Scheme
-	restConfig *rest.Config
-	recorder   record.EventRecorder
+	mgr manager.Manager
+	// client     client.Client
```
Comment (sabre1041) [inline pkg/util/reconciler.go:55]: Can we remove these comments?

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Minor comment cleanup. Otherwise good

File: pkg/util/reconciler.go (hunk 1 above)
Comment (raffaelespazzoli) [reply pkg/util/reconciler.go]: fixed

File: pkg/util/reconciler.go (hunk 2 above)
Comment (raffaelespazzoli) [reply pkg/util/reconciler.go]: fixed

**Comment by raffaelespazzoli** [comment]
please see if it's ok now.

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by davidkarlsen** [comment]
This makes it harder to use mocks as one needs to pass a Manager which has quite a number of functions. Maybe it it was kept as it was, but a new helper-constructor NewFrom(manager, recorder) which extracts the properties and puts them onto the struct. That way one achieve the same effect of ease of construction, but still supports unit-testing.

**Pull request merged**